pkg/ipeye/ipeye.go

```diff
@@ -0,0 +1,307 @@
+package ipeye
+
+import (
+	"bytes"
+	"math/rand"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/AlexxIT/go2rtc/internal/app"
+	"github.com/AlexxIT/go2rtc/pkg/core"
+	"github.com/AlexxIT/go2rtc/pkg/h264"
+	"github.com/AlexxIT/go2rtc/pkg/h264/annexb"
+	"github.com/AlexxIT/go2rtc/pkg/iso"
+	"github.com/gorilla/websocket"
+	"github.com/pion/rtp"
+	"github.com/pion/rtp/codecs"
+)
+
+type Producer struct {
+	core.Connection
+	conn *websocket.Conn
+
+	videoTrackID   uint32
+	clockRate      uint32
+	sps, pps       []byte
+	baseSet        bool
+	baseDecodeTime uint64
+}
+
+// Dial connects to ipeye WebSocket with required Origin
+func Dial(source string) (core.Producer, error) {
+	url, _ := strings.CutPrefix(source, "ipeye:")
+
+	dialer := websocket.Dialer{
+		Proxy:            http.ProxyFromEnvironment,
+		HandshakeTimeout: 10 * time.Second,
+	}
+	header := http.Header{}
+	header.Set("Origin", "https://ipeye.ru")
+
+	conn, _, err := dialer.Dial(url, header)
+	if err != nil {
+		return nil, err
+	}
+
+	prod := &Producer{
+		Connection: core.Connection{
+			ID:         core.NewID(),
+			FormatName: "ipeye",
+			Protocol:   "wss",
+			RemoteAddr: conn.RemoteAddr().String(),
+			URL:        url,
+			Transport:  conn,
+		},
+		conn: conn,
+	}
+
+	if err = prod.probe(); err != nil {
+		_ = conn.Close()
+		return nil, err
+	}
+
+	return prod, nil
+}
+
+// probe waits for init with avcC and extracts SPS/PPS
+func (p *Producer) probe() error {
+	log := app.GetLogger("ipeye")
```

Comment:
You shouldn't import app in pkg

Reply:
What logger should I use then? zerolog? Or is it better to delete logs?

Reply:
The current architecture does not allow using logs in pkg. This has not been worked out yet.

Reply:
Ok, deleted all logs